Closes #51 Add categories

nikola-mladenovic opened the pull request:
Since the number of the view controllers and view models has grown significantly, I've rearranged them in separate groups, for each app section.

Until we have category assets available in the Sunrise project, the local images will be used.

TestFlight build is being processed.

nikola-mladenovic commented on Sunrise/ViewModels/Categories/CategoriesViewModel.swift:
```diff
@@ -0,0 +1,192 @@
+        // Use hardcoded images till we get assets on categories
+        backgroundImage <~ activeCategories.producer.map { (activeCategories: [Category]) -> UIImage? in
+            guard let rootCategoryName = activeCategories.first?.name?.localizedString else { return nil }
+            switch rootCategoryName {
```
@cneijenhuis - currently we have 3 images, two specific to _Men_ and _Women_ categories, and I'm using the third one for all other categories.

cneijenhuis replied:
Yup, saw that :)

cneijenhuis commented:
When going from Category to a POP, I'm always browsing the Online Store. Is that intentional?

nikola-mladenovic commented:
That isn't intentional, working on fixing it

nikola-mladenovic commented:
Fixed ReactiveSwift related leaks on POP and PDP, updated UI, my store POP sync from home and categories tab. Also added a fix for reservation notifications.
